File: common/src/main/java/com/ssblur/scriptor/item/EngravingTool.java
package com.ssblur.scriptor.item;

import com.ssblur.scriptor.events.network.server.ChalkNetwork;
import net.minecraft.world.InteractionHand;
import net.minecraft.world.InteractionResultHolder;
import net.minecraft.world.entity.player.Player;
import net.minecraft.world.item.ItemStack;
import net.minecraft.world.level.Level;

public class EngravingTool extends Chalk {
  public EngravingTool(Properties properties) {
    super(properties);
  }

  @Override
  public InteractionResultHolder<ItemStack> use(Level level, Player player, InteractionHand interactionHand) {
    if(level.isClientSide)
      ChalkNetwork.sendChalkMessage(true);

    return InteractionResultHolder.success(player.getItemInHand(interactionHand));
  }
}

File: common/src/main/kotlin/com/ssblur/scriptor/item/tools/EngravingTool.kt
package com.ssblur.scriptor.item.tools

import com.ssblur.scriptor.network.server.ScriptorNetworkC2S
import com.ssblur.scriptor.network.server.ScriptorNetworkC2S.SendChalk
import net.minecraft.client.Minecraft
import net.minecraft.world.InteractionHand
import net.minecraft.world.InteractionResultHolder
import net.minecraft.world.entity.player.Player
import net.minecraft.world.item.ItemStack
import net.minecraft.world.level.Level
import net.minecraft.world.phys.BlockHitResult

class EngravingTool(properties: Properties): Chalk(properties) {
  override fun use(
    level: Level,
    player: Player,
    interactionHand: InteractionHand
  ): InteractionResultHolder<ItemStack> {
    if (level.isClientSide) {
      val client = Minecraft.getInstance()
      val hit = client.hitResult
      if (hit is BlockHitResult)
        ScriptorNetworkC2S.sendChalk(SendChalk(hit, true))
    }

    return InteractionResultHolder.success(player.getItemInHand(interactionHand))
  }
}
